src/include/duckdb/function/table/range/range_executor.hpp

```diff
@@ -0,0 +1,61 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/function/table/range/range_executor.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/execution/expression_executor.hpp"
+#include "duckdb/planner/expression/bound_reference_expression.hpp"
+
+namespace duckdb {
+
+class RangeExecutor {
+protected:
+	RangeExecutor(ClientContext &context, const vector<unique_ptr<Expression>> &args_list, bool generate_series_p)
+	    : first_fetch(true), generate_series(generate_series_p), expr_executor(context) {
+
+		// we add each expression in the args_list to the expression executor
+		vector<LogicalType> args_data_types;
+		for (auto &exp : args_list) {
+			D_ASSERT(exp->type == ExpressionType::BOUND_REF);
+			auto &ref = exp->Cast<BoundReferenceExpression>();
+			args_data_types.push_back(exp->return_type);
+			expr_executor.AddExpression(ref);
+		}
+
+		auto &allocator = Allocator::Get(context);
+		args_data.Initialize(allocator, args_data_types);
+	}
+
+public:
+	~RangeExecutor() {
```

Comment:
This needs to be marked `virtual` because at least one of the methods is marked `virtual`